### ISSUE TITLE
AWS ECS Fargate Docker image changes not recognized

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi-cloud#595]
_From @johscheuer on September 14, 2018 7:6_

I followed the steps described here: https://pulumi.io/quickstart/aws/tutorial-containers-ecs-fargate.html

The following snippets works fine for the initial setup of ECS/Fargate.
```javascript
const cloud = require("@pulumi/cloud-aws");

let service = new cloud.Service("pulumi-nginx", {
    containers: {
        nginx: {
            build: "./app",
            memory: 128,
            ports: [{ port: 80 }],
        },
    },
    replicas: 2,
});

// export just the hostname property of the container frontend
exports.url = service.defaultEndpoint.apply(e => `http://${e.hostname}`);
```
When I change the Docker image and rebuild it with `pulumi update` I see the newly built Docker image in ECR but the ECS/Fargate Task won't be update. Is this a wanted behaviour? I would expect that the newly built Docker image will be deployed on AWS Fargate.

If I use an image e.g. with `image: "nginx:1.15.3"` instead of the `build` definition everything works as expected.

_Copied from original issue: pulumi/pulumi-aws#319_

**Comment by lukehoban** [comment]
Thanks for the report.  This is a fairly serious regression in `0.15.x` of `@pulumi/cloud`.

I confirmed that this works as expected with `0.14.2`.

The root cause is that the TaskDefinition is getting the following environment variable added - missing the actual digest hash that would force the update:

```
    {
          "name": "IMAGE_DIGEST",
          "value": ""
        },
```

**Comment by CyrusNajmabadi** [comment]
Looking now.

**Comment by lukehoban** [comment]
This appears to be due to the changes in https://github.com/pulumi/pulumi-docker/pull/10.  The logic to extract Docker image digests using the `docker` CLI does not behave correctly in the model used by pulumi-cloud where the locally built image does not have the same name as the target image (and is not tagged until after we search for the digest).

I'll see if I can pull together a fix.

**Comment by lukehoban** [comment]
This was fixed in https://github.com/pulumi/pulumi-docker/pull/17.  That fix is released with `@pulumi/docker` 0.15.2, so updating dependencies should give you the fix, and fresh installs of latest `@pulumi/cloud` should get the fix.
